x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/analytics/action/PutAnalyticsCollectionAction.java

```diff
@@ -0,0 +1,109 @@
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.entsearch.analytics.AnalyticsCollection;
+import org.elasticsearch.xpack.entsearch.analytics.action.support.BaseAnalyticsCollectionResponse;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+public class PutAnalyticsCollectionAction extends ActionType<PutAnalyticsCollectionAction.Response> {
+
+    public static final PutAnalyticsCollectionAction INSTANCE = new PutAnalyticsCollectionAction();
+    public static final String NAME = "cluster:admin/behavioral_analytics/put";
+
+    public PutAnalyticsCollectionAction() {
+        super(NAME, PutAnalyticsCollectionAction.Response::new);
+    }
+
+    public static class Request extends MasterNodeRequest<Request> {
+        private final AnalyticsCollection analyticsCollection;
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.analyticsCollection = new AnalyticsCollection(in);
+        }
+
+        public Request(String resourceName, BytesReference content, XContentType contentType) {
+            this.analyticsCollection = AnalyticsCollection.fromXContentBytes(resourceName, content, contentType);
+        }
+
+        public Request(AnalyticsCollection analyticsCollection) {
+            this.analyticsCollection = analyticsCollection;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            ActionRequestValidationException validationException = null;
+
+            final String name = analyticsCollection.getName();
+
+            if (name == null || name.isEmpty()) {
+                validationException = addValidationError("Analytics collection name is missing", validationException);
+            }
+
+            return validationException;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            analyticsCollection.writeTo(out);
+        }
+
+        public AnalyticsCollection getAnalyticsCollection() {
+            return analyticsCollection;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Request that = (Request) o;
+            return Objects.equals(analyticsCollection, that.analyticsCollection);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(analyticsCollection);
+        }
+    }
+
+    public static class Response extends BaseAnalyticsCollectionResponse implements ToXContentObject {
```

Comment:
For Engines, the  PUT API is returning just "result": "created" or "updated". This is probably more similar to what other ES APIs do today.

Do you have a use case for returning the whole analytics collection information?

Reply:
ATM the design has been like for like on the previous API. We were originally required the whole model back for updates so we refresh the client model without needing to make another call to GET endpoint. That update mechanism will likely change now we are using datastreams directly. 

Will leave it for now as model but possible change in contract when we get into kibana integration detail.